### PR TITLE
Diann parameters

### PR DESCRIPTION
**Pull request by ArtiVlasov**

### CODE REVIEW

File: jsons/parameters.json [hunk 1]
```diff
@@ -10243,8 +10389,9 @@
     },
     {
         "default_value": 150,
-        "description": " lowest considered mz for quantification ",
+        "description": "Defines the minimum m/z value to be considered by the respective tool.",
```
Comment (fu) [inline jsons/parameters.json:10392]: <3

File: jsons/parameters.json [hunk 2]
```diff
@@ -14539,26 +14842,60 @@
         "value_type": "bool"
     },
     {
-        "default_value": {},
-        "description": "Defines the column names which should have their different values merged into a single value when merging rows corresponding the same PSM Formatted as a dictionary with keys as the column names and values as a parameter to specify which one of the different values to take Available values:    max_value\n    min_value\n    most_frequent\n    avg_value",
+        "default_value": "heuristic",
+        "description": "Protein inference algorithm that DIANN should use.",
```
Comment (fu) [inline jsons/parameters.json:14846]: Not sure if DIANN is the right thing in this description

File: jsons/parameters.json [hunk 3]
```diff
@@ -16750,6 +17386,40 @@
         },
         "value_type": "bool"
     },
+    {
+        "default_value": true,
+        "description": "Either to use unimod4. TODO: this param need revisiting, once it is clear what the param stand for!",
+        "key_translations": {
+            "diann_style_1": "use_unimod4_<DROP_KEY>",
+            "ursgal_style_1": "use_unimod4"
```
Comment (fu) [inline jsons/parameters.json:17394]: can unimod not be supplied to the algorithm? If not what about usermods? If so then we should not have a parameters for that but let the wrapper decide to use unimod4 if no unimod file has been applied, I guess.

**Review by fu — CHANGES_REQUESTED** [review]
See comments

File: jsons/parameters.json (hunk 3 above)
Comment (ArtiVlasov) [reply jsons/parameters.json:17394]: Both Andrew and me don't know what this param is actually doing. I added it for now as it is part of the command line. During testing, Andrew wanted to evaluate this and then update the param or even get rid of it. Hence, the 'TODO' in the description :)

Comment (andy-d-palmer) [reply jsons/parameters.json:17394]: From the source it looks like this is a shorthand for adding unimods by accession, diann ships with the 'full unimod' ` --full-unimod loads the complete UniMod modification database and disables the automatic conversion of modification names to the UniMod format` there's also the documented --mod argument `--mod [name],[mass],[optional: 'label'] declares a modification name. Examples: "--mod UniMod:5,43.005814", "--mod SILAC-Lys8,8.014199,label"`.

I would be fine with dropping the shorthand and pushing mods through the command line arguments.

File: jsons/parameters.json [hunk 4]
```diff
@@ -15105,6 +15476,40 @@
...
+                    false,
+                    false
+                ],
+                [
+                    true,
+                    true
+                ]
+            ]
+        },
+        "value_type": "bool"
```
Comment (andy-d-palmer) [inline jsons/parameters.json:15511]: I know it's not properly documented on the command line reference but this variable is a qvalue filter and takes a float value.

Comment (ArtiVlasov) [reply jsons/parameters.json:15511]: so it rather has the shape `--qvalue 0.1`? I just took the flag from the reporter repo, as it did not had any value behind the param ^^

**Review by MKoesters — APPROVED** [review]

**Pull request merged**